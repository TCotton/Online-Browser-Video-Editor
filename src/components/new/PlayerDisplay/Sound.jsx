import React, {useState} from "react";
import {useDispatch} from "react-redux";
import {soundFn} from "../slices/playerSlice";

export const Sound = () => {
    const dispatch = useDispatch();
    const [mute, setMute] = useState(false);
    const onClick = () => {
        setMute(!mute);
        dispatch(soundFn());
    }
    return (
        <img src={mute ? "./mute.svg" : "./sound.svg"} className="sound" alt={mute ? "mute" : "sound"}
             onClick={onClick} data-testid="sound"/>
    )
}