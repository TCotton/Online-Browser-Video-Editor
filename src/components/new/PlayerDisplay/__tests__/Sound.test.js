import React from 'react';
import { render } from '@testing-library/react';
import { Sound } from '../Sound.jsx';

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn()
}));
 
describe('Sound', () => {

    it('should render successfully', () => {
        const {container} = render(<Sound />);
        expect(container.querySelector("[data-testid=\"sound\"]")).not.toBeNull();
    });

    it('should match snapshot', () => {
        const {baseElement} = render(<Sound />);
        expect(baseElement).toMatchSnapshot();
    });

});