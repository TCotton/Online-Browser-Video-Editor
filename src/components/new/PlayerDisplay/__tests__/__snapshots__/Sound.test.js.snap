// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sound should match snapshot 1`] = `
<body>
  <div>
    <img
      alt="sound"
      class="sound"
      data-testid="sound"
      src="./sound.svg"
    />
  </div>
</body>
`;
